File: Zad3.cpp
#include <iostream>
using namespace std;

class Car {
public:
	string make;
	string model;
	int year;

	int currentYear = 2024;

	Car(string make, string model, int year) {
		this->make = make;
		this->model = model;
		this->year = year;
	}

	void displayInfo() {
		cout << make << " " << model << " " << year << endl;
	}

	int getCarAge() {
		return currentYear - year; 
	}
	
	~Car(){}
};

bool isSameCar(Car car1, Car car2) {
	if (car1.make == car2.make && car1.model == car2.model && car1.year == car2.year) {
		return true;
	}
	else {
		return false;
	}
}

int main() {
	int wiek;
	Car samochod1("Fiat", "500", 2017);
	Car samochod2("Fiat", "500", 2017);
	Car samochod3("BMW", "X5", 2022);
	samochod1.displayInfo();
	wiek = samochod1.getCarAge();
	cout << wiek << endl;
	cout << (isSameCar(samochod1, samochod2) ? "Takie same" : "Rozne") << endl;
	cout << (isSameCar(samochod1, samochod3) ? "Takie same" : "Rozne") << endl;
	return 0;
}

File: Zad3.java
package test;

public class Car {
	
	String make;
	String model;
	int year;
	
	int currentYear = 2024;

	public Car(String make, String model, int year) {
		this.make = make;
		this.model = model;
		this.year = year;
	}
	
	void displayInfo() {
		System.out.println(make + " " + model + " " + year);
	}
	
	int getCarAge() {
		return currentYear - year; 
	}
	
    boolean isSameCar(Car otherCar) {
        return this.make.equals(otherCar.make) && 
               this.model.equals(otherCar.model) && 
               this.year == otherCar.year;
    }
	
	public static void main(String[] args) {
		int wiek;
		
		Car samochod1 = new Car("Fiat", "500", 2017);
		Car samochod2 = new Car("Fiat", "500", 2017);
		
		samochod1.displayInfo();
		samochod2.displayInfo();
		
		wiek = samochod1.getCarAge();
		System.out.println(wiek);
		
		boolean wynik = samochod1.isSameCar(samochod2);
        System.out.println("Czy auta sa takie same: " + wynik);
	}

}